libs/facilities/backend/timeseries/src/lib/dto/request.dto.ts

```diff
@@ -0,0 +1,81 @@
+import { Type } from 'class-transformer';
+import {
+	IsBoolean,
+	IsDate,
+	IsDateString,
+	IsEnum,
+	IsNotEmpty,
+	IsNumber,
+	IsOptional,
+	IsString,
+} from 'class-validator';
+
+import { IGetTimeSeriesParams, IGetTimeseriesQuery } from '../interfaces/request.interface';
+
+/**
+ * The DTO for the time series parameters,
+ * this is to validate the path parameters.
+ */
+export class GetTimeSeriesParamsDto implements IGetTimeSeriesParams {
+	/**
+	 * The entity id for an asset
+	 */
+	@IsNotEmpty()
+	@IsString()
+	entityId: string;
+
+	/**
+	 * The name of the aspect
+	 */
+	@IsNotEmpty()
+	@IsString()
+	propertySetName: string;
+}
+
+/**
+ * The DTO for the time series query,
+ * this is to validate the query parameters.
+ */
+export class GetTimeSeriesQueryDto implements IGetTimeseriesQuery {
+	/**
+	 * The start date for the time series data
+	 */
+	@IsOptional()
+	@IsDate()
+	@Type(() => Date)
```

Comment:
Just @IsType is enough

Reply:
The @Type here needs to be done to cast into date